example/aws_iot/home_temp_simulated.py

```diff
@@ -0,0 +1,108 @@
+#  Redistribution and use in source and binary forms, with or without         #
+#  modification, are permitted provided that the following conditions are met:#
+#                                                                             #
+#  - Redistributions of source code must retain the above copyright notice,   #
+#      this list of conditions and the following disclaimer.                  #
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+
+from liota.dcc.aws_iot import AWSIoT
+from liota.identity.gateway_identity import Identity
+from liota.boards.gateway_dk300 import Dk300
+from AWSIoTPythonSDK.MQTTLib import AWSIoTMQTTClient
+import random
+
+# getting values from conf file
+config = {}
+execfile('../sampleProp.conf', config)
+
+
+# Random number generator, simulating living room temperature readings.
+def living_room_temperature():
+    return random.randint(10, 30)
+
+
+# Random number generator, simulating kitchen temperature readings.
+def kitchen_temperature():
+    return random.randint(10, 30)
+
+
+# Random number generator, simulating bedroom temperature readings.
+def bedroom_temperature():
+    return random.randint(10, 30)
+
+
+if __name__ == '__main__':
+    gateway = Dk300(config['Gateway1Name'])
+
+    #  Using Identity layer for holding details related to authentication
+    identity = Identity(config['RootCAPath'], config['ClientCertPath'], config['PrivateKeyPath'], None, None)
+
+    #  Connecting to AWS
+    #  These are standard steps to follow to connect with AWS IoT using AWS IoT SDK
+    mqtt_client = AWSIoTMQTTClient(config['ClientName'])
```

Comment:
I guess by defining a common DCC layer, we are trying to avoid these exact steps. All these operations can be performed within AWSIoT.__init__() method so that the end client code is simplified.